backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java

```diff
@@ -54,19 +61,133 @@ public Response setCommit(
     }
   }
 
+  @Override
+  public Response setCommitFromDatasetVersion(
+      DatasetVersion datasetVersion,
+      DatasetDAO datasetDAO,
+      BlobDAO blobDAO,
+      RepositoryDAO repositoryDAO,
+      MetadataDAO metadataDAO,
+      FileHasher fileHasher)
+      throws ModelDBException, NoSuchAlgorithmException {
+    try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+      session.beginTransaction();
+      DatasetBlob.Builder datasetBlobBuilder = DatasetBlob.newBuilder();
+      Blob.Builder blobBuilder = Blob.newBuilder();
+      blobBuilder.addAllAttributes(datasetVersion.getAttributesList());
+      switch (datasetVersion.getDatasetVersionInfoCase()) {
+        case RAW_DATASET_VERSION_INFO:
+        case QUERY_DATASET_VERSION_INFO:
+          throw new ModelDBException("Not supported", Code.UNIMPLEMENTED);
```

Comment:
we can remove these from protos itself, https://github.com/VertaAI/modeldb/blob/4375484f4a24f13af610c804f3a3b0344592498e/protos/protos/public/modeldb/DatasetVersionService.proto#L29-L50

Reply:
But QueryDatasetVersionInfo is used in QueryDatasetVersionInfoEntity which is used in many places. Is it necessary to delete it now? Can it be a separate task?

Reply:
yes we can have it as different task